linker/linker.cpp

```diff
@@ -649,6 +655,8 @@ class LoadTask {
     si_->load_bias = elf_reader.load_bias();
     si_->phnum = elf_reader.phdr_count();
     si_->phdr = elf_reader.loaded_phdr();
+    si_->rand_addr_segments = elf_reader.rand_addr_segments();
+    si_->sort_rand_addr_segments();
```

Comment:
I've put the sort inline in a new `set_rand_addr_segments()` function and commented on the sort order.

I don't think we can have a `const` member vector. That would have to be initialized in the `soinfo` constructor, and we're not doing that (see `LoadTask`).

Plain old C arrays are (almost) always a bad idea. When possible, avoid.